### PR TITLE
Singleton

### PR DESCRIPTION
**Pull request by segerphilip**
Adds an Application class that can be used as a singleton for location or playlogin objects. Also, bugfix, hides the login button on success.

I know that I said I'd make a singleton, but this turns out to be easier to do for the Play Login stuff.

On second thought, I don't know why I did this without consulting Sam.

### CODE REVIEW

File: app/src/main/java/website/bloop/app/BloopApplication.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private GoogleApiClient mGoogleApiClient;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        mInstance = this;
+    }
+
+    public static BloopApplication getInstance(){
+        if(mInstance == null)
```
Comment (wolfd) [inline app/src/main/java/website/bloop/app/BloopApplication.java]: small nit: keep the starting bracket on the same line as the if, and have a space after the `if` and before the `(`.

**Review by wolfd — CHANGES_REQUESTED** [review]
Looks great.

Did you know that for some reason, we can edit each other's comments? That's not going to go badly.

Yes Danny. Yes it will.

File: app/src/main/java/website/bloop/app/BloopApplication.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    private GoogleApiClient mGoogleApiClient;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        mInstance = this;
+    }
+
+    public static BloopApplication getInstance(){
```
Comment (sammyers) [inline app/src/main/java/website/bloop/app/BloopApplication.java]: Yo where's that space.

where's it @

File: app/src/main/java/website/bloop/app/BloopApplication.java (hunk 1 above)
Comment (sammyers) [reply app/src/main/java/website/bloop/app/BloopApplication.java]: savage

Comment (segerphilip) [reply app/src/main/java/website/bloop/app/BloopApplication.java]: oh man how did I miss this sorry

**Review by wolfd — APPROVED** [review]
A+

**Pull request merged**